File: q0028/q28.java
class Solution {
    public int strStr(String haystack, String needle) {
        int l = needle.length();
        if (l > haystack.length()) {
            return -1;
        }
        for (int i = 0; i < haystack.length() - l + 1; i++) {
            if (haystack.substring(i, i + l).equals(needle)) {
                return i;
            }
        }
        return -1;
    }
}

File: q0028/q28.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        l = len(needle)
        if l > len(haystack):
            return -1
        for i in range(len(haystack) - l + 1):
            if haystack[i : i + l] == needle:
                return i
        return -1
